src/contexts/EthereumContext.tsx

```diff
@@ -207,13 +222,62 @@ export function EthereumProvider({ children }: { children: React.ReactNode }) {
   // Disconnect the provider by resetting the app's account
   const disconnect = () => {
     setAccount(null)
+    setVerified(false)
   }
 
   // Toggle function
   const toggleOnboard = () => {
     setUseOnboard(!useOnboard)
   }
 
+  // Sign In With Ethereum
+  const signInWithEthereum = async () => {
+    // SIWE requires an connected account
+    if (!account || !provider) {
+      console.log('No account connected')
+      return
+    }
+
+    // Get chain ID of provider.
+    const { chainId } = await provider.getNetwork()
+
+    const siweMessage = new SiweMessage({
+      domain: window.location.host,
+      address: account,
+      statement: 'By logging in you agree to the terms and conditions.',
+      uri: window.location.origin,
+      version: '1',
+      chainId: Number(chainId),
+      resources: ['https://boilerplate.lukso.tech'],
+    })
+
+    const message = siweMessage.prepareMessage()
+    console.log('firstmessage:', message)
+    const hashedMessage = ethers.hashMessage(message)
+
+    try {
+      const signer = await provider.getSigner()
+      const signature = await signer.signMessage(message)
+
+      // Create the UniversalProfile contract instance
+      const myUniversalProfileContract = new ethers.Contract(
+        account,
+        UniversalProfileContract.abi,
+        provider
+      )
+
+      const isValidSignature =
+        await myUniversalProfileContract.isValidSignature(
+          hashedMessage,
+          signature
+        )
+
+      setVerified(isValidSignature === '0x1626ba7e')
```

Comment:
Please put `0x1626ba7e` in a variable so we can name it and reference to it:

```
// https://eips.ethereum.org/EIPS/eip-1271
const MAGIC_VALUE = "0x1626ba7e";
```

It makes the code easier to read